.github/workflows/ci.yml

```
@@ -8,35 +8,55 @@ on:
     branches:
       - "main"
       - "master"
+
 jobs:
-  test:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
+  run:
 
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
-        with:
-          python-version: "3.10"
+    runs-on: ${{ matrix.config.os }}
 
-      - name: Install dependencies
-        # https://docs.github.com/en/free-pro-team@latest/actions/guides/building-and-testing-python#installing-dependencies
-        # If installing a development version of constellation, use:
-        # - pip3 install git+https://github.com/reside-ic/constellation@reside-62#egg=constellation
-        run: |
-          python -m pip install --upgrade setuptools pip wheel
-          pip3 install pytest-cov pycodestyle codecov
-          pip3 install -r requirements.txt
+    name: ${{matrix.config.os }} (${{ matrix.config.py }})
+
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {os: ubuntu-latest,  py: '3.8'}
+          - {os: ubuntu-latest,  py: '3.9'}
+          - {os: ubuntu-latest,  py: '3.10'}
+          - {os: ubuntu-latest,  py: '3.11'}
+          - {os: ubuntu-latest,  py: '3.12'}
 
-      - name: Tests
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.config.py }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.config.py }}
+      - name: Pull containers
+        run: |
+          docker pull redis:5.0
+          docker pull vimc/orderly.server:master
+          docker pull vimc/orderly-web:master
+          docker pull vimc/orderly-web-proxy:master
+          docker pull vimc/orderly-web-css-generator:master
+          docker pull vimc/orderly-web-user-cli:master
+          docker pull mrcide/outpack.orderly:main
+          docker pull mrcide/outpack_server:main
+          docker pull mrcide/packit-db:main
+          docker pull mrcide/packit-api:main
+          docker pull mrcide/montagu-packit:mrc-5140-trusted-headers
```

We'll have to remember to update this branch at some point

---

This has been merged to main so this can be 
```suggestion
          docker pull mrcide/montagu-packit:main
```